subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractModuleComponentResolveMetadata.java

```diff
@@ -135,6 +137,19 @@ public ModuleSource getSource() {
     }
 
     @Override
+    public List<? extends ConfigurationMetadata> getConsumableConfigurationsHavingAttributes() {
+        if (consumableConfigurations == null) {
```

Comment:
This is a view (subset of) configurations. Do you mean that:
- the field should not be lazily initialized (I wanted to avoid initializing in the c'tor since it might not always be needed)
- or that it should return an immutable list?
- or both?